presto-hive/src/main/java/io/prestosql/plugin/hive/util/HiveBucketing.java

```diff
@@ -203,21 +204,35 @@ static int getBucketNumber(int hashCode, int bucketCount)
         return Optional.of(new HiveBucketHandle(bucketColumns.build(), bucketingVersion, bucketCount, bucketCount));
     }
 
-    public static Optional<HiveBucketFilter> getHiveBucketFilter(Table table, TupleDomain<ColumnHandle> effectivePredicate)
+    public static Optional<HiveBucketFilter> getHiveBucketFilter(HiveTableHandle hiveTable, TupleDomain<ColumnHandle> effectivePredicate)
     {
-        if (table.getStorage().getBucketProperty().isEmpty()) {
+        if (hiveTable.getBucketHandle().isEmpty() || effectivePredicate.isAll()) {
```

Comment:
check `effectivePredicate.isAll()` should go to the method below

Reply:
i got rid of this and keep the code flow similar to the current one